ios/FluentUI/Core/Theme/Tokens/FontInfo.swift

```diff
@@ -17,47 +17,75 @@ public struct FontInfo {
         self.size = size
         self.weight = weight
     }
 
+    static var sizeTuples: [(size: CGFloat, textStyle: Font.TextStyle)] = [
+        (34.0, .largeTitle),
+        (28.0, .title),
+        (22.0, .title2),
+        (20.0, .title3),
+        // Note: `17.0: .headline` is removed to avoid needing duplicate size key values.
+        // But it's okay because Apple's scaling curve is identical between it and `.body`.
+        (17.0, .body),
+        (16.0, .callout),
+        (15.0, .subheadline),
+        (13.0, .footnote),
+        (12.0, .caption),
+        (11.0, .caption2)
+    ]
+
+    public var textStyle: Font.TextStyle {
+        // Defaults to smallest supported text style for mapping, before checking if we're bigger.
+        var textStyle = Font.TextStyle.caption2
+        for tuple in Self.sizeTuples {
+            if self.size >= tuple.size {
+                textStyle = tuple.textStyle
+                break
+            }
+        }
+        return textStyle
+    }
+
     let name: String?
     let size: CGFloat
     let weight: Font.Weight
 }
 
 // MARK: - ViewModifier
 
-extension Font {
+public extension Font {
     static func fluent(_ fontInfo: FontInfo, shouldScale: Bool = true) -> Font {
-        let size = shouldScale ?
-            UIFontMetrics.default.scaledValue(for: fontInfo.size) :
-            fontInfo.size
-
-        let font: Font
-        if let name = fontInfo.name {
-            // We use fixedSize because scaling is already managed above.
-            font = .custom(name, fixedSize: size)
-        } else {
-            font = .system(size: size)
-        }
-        return font.weight(fontInfo.weight)
+        // SwiftUI Font is missing some of the ease of construction available in UIFont.
+        // So just leverage the logic there to create the equivalent SwiftUI font.
+        let uiFont = UIFont.fluent(fontInfo, shouldScale: shouldScale)
+        return Font(uiFont)
     }
 }
 
 extension UIFont {
-    static func fluent(_ fontInfo: FontInfo, shouldScale: Bool = true) -> UIFont {
-        let size = shouldScale ?
-            UIFontMetrics.default.scaledValue(for: fontInfo.size) :
-            fontInfo.size
+    public static func fluent(_ fontInfo: FontInfo, shouldScale: Bool = true) -> UIFont {
+        let unscaledFont: UIFont
 
         if let name = fontInfo.name,
-           let font = UIFont(name: name, size: size) {
-            return font.withWeight(uiWeight(fontInfo.weight))
+           let font = UIFont(name: name, size: fontInfo.size) {
+            // Named font
+            unscaledFont = font.withWeight(uiWeight(fontInfo.weight))
         } else {
-            return .systemFont(ofSize: size, weight: uiWeight(fontInfo.weight))
+            // System font
+            unscaledFont = .systemFont(ofSize: fontInfo.size, weight: uiWeight(fontInfo.weight))
+        }
+
+        if shouldScale {
+            let fontMetrics = UIFontMetrics(forTextStyle: uiTextStyle(fontInfo.textStyle))
```

Comment:
I can't seem to find it in the documentation, and it probably does / seems to, but does this maintain the weight that we pass in? The documentation just sort of mentions the base font and doesn't explicitly say that it'll preserve the weight of that font.

Reply:
Yes it does -- the scaled typography in the attached screenshots maintain their weights. Without the posted weights, those fonts would look _much_ less bold.